src/python/pants/option/global_options.py

```diff
@@ -692,21 +695,43 @@ def register_bootstrap_options(cls, register):
             advanced=True,
             type=bool,
             default=DEFAULT_EXECUTION_OPTIONS.remote_execution,
-            help="Enables remote workers for increased parallelism. (Alpha)",
+            help=(
+                "Enables remote workers for increased parallelism. (Alpha)\n\nAlternatively, you "
+                "can use `--remote-cache-read` and `--remote-cache-write` to still run everything "
+                "locally, but to use a remote cache."
+            ),
         )
         register(
             "--remote-cache-read",
             type=bool,
             default=DEFAULT_EXECUTION_OPTIONS.remote_cache_read,
             advanced=True,
-            help="Whether to enable reading from a remote cache",
+            help=(
+                "Whether to enable reading from a remote cache.\n\nThis cannot be used at the same "
+                "time as `--remote-execution`."
+            ),
         )
         register(
             "--remote-cache-write",
             type=bool,
             default=DEFAULT_EXECUTION_OPTIONS.remote_cache_write,
             advanced=True,
-            help="Whether to enable writing results to a remote cache",
+            help=(
+                "Whether to enable writing results to a remote cache.\n\nThis cannot be used at "
+                "the same time as `--remote-execution`."
+            ),
+        )
+        register(
+            "--remote-cache-eager-fetch",
+            type=bool,
+            advanced=True,
+            default=DEFAULT_EXECUTION_OPTIONS.remote_cache_eager_fetch,
+            help=(
+                "Eagerly fetch from the remote store instead of lazily fetching.\n\nThis results "
+                "in worse performance, but reduces the frequency of errors encountered.\n\nThis is "
+                "meant to be a short-term mechanism. As we make Pants's remote caching client more "
+                "robust, we plan to remove this mechanism."
```

Comment:
@benjyw conjectured this may actually be faster in some environments like desktop because we reduce the # of trips.

Reply:
And even if his conjecture is wrong, isn't it correct to say the converse is also a conjecture? If so the definitive language in the help should go.

Reply:
Yeah, I wouldn't commit in the help to this being short term. Can just say "this may provide more robust behavior and/or better performance".